macros/src/lib.rs

```diff
@@ -2,8 +2,234 @@ extern crate proc_macro;
+                    };
+
+                    let enum_mut_index = enum_index.clone();
+
+                    let fields_draw = field_idents.iter().map(|f| quote!(#f.draw(display)?;));
+                    let enum_draw = quote! {
+                        Self::#variant_name { #(#field_idents,)* } => {
+                            #(#fields_draw)*
+                        }
+                    };
+
+                    (
+                        enum_field_count,
+                        enum_translate,
+                        enum_index,
+                        enum_mut_index,
+                        enum_draw,
+                    )
+                }
+                Fields::Unnamed(FieldsUnnamed { unnamed, .. }) => {
+                    let field_idents = unnamed
+                        .iter()
+                        .enumerate()
+                        .map(|(num, _)| format_ident!("__self_{}", num))
+                        .collect::<Vec<_>>();
+
+                    let fields_count = unnamed.iter().count();
+                    let enum_field_count = quote! {
+                        Self::#variant_name(#(#field_idents,)*) => {
+                            #fields_count
+                        }
+                    };
+
+                    let translate_fields = field_idents.iter().map(|f| quote!(#f.clone().translate(by), ));
+                    let enum_translate = quote! {
+                        Self::#variant_name(#(#field_idents),*) => {
+                            Self::#variant_name(
+                                #(#translate_fields)*
+                            )
+                        }
+                    };
+
+                    let fields_index = field_idents
+                        .iter()
+                        .enumerate()
+                        .map(|(i, f)| quote!(#i => #f,));
+                    let enum_index = quote! {
+                        Self::#variant_name(#(#field_idents),*) => {
+                            match index {
+                                #(#fields_index)*
+                                _ => panic!("Unknown index!"),
+                            }
+                        }
+                    };
+
+                    let enum_mut_index = enum_index.clone();
+
+                    let field_draws = field_idents.iter().map(|f| quote!(#f.draw(display)?;));
+                    let enum_draw = quote! {
+                        Self::#variant_name(#(#field_idents),*) => {
+                            #(#field_draws)*
+                        }
+                    };
+
+                    (
+                        enum_field_count,
+                        enum_translate,
+                        enum_index,
+                        enum_mut_index,
+                        enum_draw,
+                    )
+                }
+                Fields::Unit => {
+                    panic!("Can't have unit enum. Contents of enum must impl View + Drawable")
```

Comment:
Hmm, it might make sense to support units, I think, in case you don't want to display anything for a particular case. Is there any difficulty here besides generating code that does nothing?

Reply:
@bugadani I didn't address this in the code yet. Was waiting to see what you had to say about how to implement a view that has a no-op draw function.

Reply:
> or the library will [crash](https://github.com/bugadani/embedded-layout/blob/888d57129096f7e8b2166c26452d3269ec791a69/src/view_group/mod.rs#L39)

Oh nevermind then, I'll do this later.